docs/hazmat/primitives/asymmetric/serialization.rst

```diff
@@ -309,6 +309,25 @@ Serialization Formats
         encryption. Choose this unless you have explicit legacy compatibility
         requirements.
 
+.. class:: PublicFormat
+
+    .. versionadded:: 0.8
+
+    An enumeration for public key formats. Used with the ``public_bytes``
+    method available on
+    :class:`~cryptography.hazmat.primitives.asymmetric.rsa.RSAPublicKeyWithSerialization`.
+
+    .. attribute:: SubjectPublicKeyInfo
+
+        This is the typical public key format. It consists of an algorithm
+        identifier and the public key as a bit string. Choose this unless
+        you have specific needs.
+
+    .. attribute:: PKCS1
+
+        Just the public key elements (without the algorithm identifier). This
+        format is occasionally used by various systems.
```

Comment:
Let's add that to the docs then.

On Sun, Mar 8, 2015 at 3:08 PM, Paul Kehrer <notifications@github.com>
wrote:

> In docs/hazmat/primitives/asymmetric/serialization.rst
> <https://github.com/pyca/cryptography/pull/1706#discussion_r26007880>:
>
> > +    .. versionadded:: 0.8
> > +
> > +    An enumeration for public key formats. Used with the ``public_bytes``
> > +    method available on
> > +    :class:`~cryptography.hazmat.primitives.asymmetric.rsa.RSAPublicKeyWithSerialization`.
> > +
> > +    .. attribute:: SubjectPublicKeyInfo
> > +
> > +        This is the typical public key format. It consists of an algorithm
> > +        identifier and the public key as a bit string. Choose this unless
> > +        you have specific needs.
> > +
> > +    .. attribute:: PKCS1
> > +
> > +        Just the public key elements (without the algorithm identifier). This
> > +        format is occasionally used by various systems.
>
> Pretty much. Probably worth documenting that explicitly for now (DSA does
> exist but OpenSSL doesn't expose it as a macro so by default you can't load
> it unless you use some undocumented functions).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1706/files#r26007880>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084